### PR TITLE
fix: normalize Deno.execPath

### LINKED ISSUE #1798: Unneeded path segments in `Deno.execPath`
```
$ ./target/debug/deno
> Deno.execPath
/Users/kevinqian/Desktop/Programming/Deno/deno/./target/debug/deno
```
Confusing to have an extra `/./` in the path
/cc @sh7dm

### PR DESCRIPTION
**Pull request by bartlomieju**
Fixes #1798

### CODE REVIEW

**Comment by bartlomieju** [comment]
I'm wondering if using [`components`](https://doc.rust-lang.org/stable/std/path/struct.Path.html#method.components) would be better solution for Path normalization than using Url parser. `Components` indeed removes `.` from paths (`a/./b` -> `a/b`) but leaves `..` because they may be symlinks.

File: cli/ops.rs [hunk 1]
```diff
@@ -345,8 +346,16 @@ fn op_start(
   let cwd_off =
     builder.create_string(deno_fs::normalize_path(cwd_path.as_ref()).as_ref());
 
+  let current_exe = std::env::current_exe().unwrap();
+  // Now apply URL parser to current exe to get fully resolved path, otherwise we might get
+  // `./` and `../` bits in `exec_path`:
+  // $ ./target/debug/deno
+  // > Deno.execPath
+  // /dev/deno/./target/debug/deno
```
Comment (ry) [inline cli/ops.rs]: The above three lines are superfluous.

**Comment by ry** [comment]
Hmm actually the test that I provided doesn't fail on master...

**Comment by bartlomieju** [comment]
> Hmm actually the test that I provided doesn't fail on master...

That's because we create absolute path to Deno executable in test suite.

**Comment by bartlomieju** [comment]
@ry this PR should be landable now

File: tests/exec_path.ts [hunk 2]
```diff
@@ -1 +1,27 @@
...
+  const textOutput = new TextDecoder().decode(output);
+
+  if (textOutput.indexOf("/./") > -1) {
+    throw Error("Exec path contains non-normalized components");
+  }
+}
+
+if (Deno.platform.os !== "win") {
+  main();
+}
```
Comment (ry) [inline tests/exec_path.ts]: This test is pretty complicated, and I don't like that it's not runnable outside the test.py framework.. (I tried running it manually on my computer, and it failed)

This patch is a fairly minor change. I would say let's just land it without tests...

**Pull request merged**